File: multiplayer-gaming-portal/src/app/auth/models/user.model.ts
// Mirrors User definition from backend (auth-service/src/types/auth.types.ts and openapi.yaml)

export interface User {
  id: string;
  email: string;
  phone?: string | null;
  status: 'pending_verification' | 'active' | 'suspended' | 'banned';
  phone_verified?: boolean;
  is_otp_enabled?: boolean;
  created_at?: string | Date; // Allow string for JSON, Date for internal use
  updated_at?: string | Date;
}

// Helper function to convert date strings to Date objects if needed
export function parseUserDates(user: User): User {
  return {
    ...user,
    created_at: user.created_at ? new Date(user.created_at) : undefined,
    updated_at: user.updated_at ? new Date(user.updated_at) : undefined,
  };
}

File: flutter_mobile_app/lib/models/user_model.dart
import 'package:flutter/foundation.dart';

@immutable
class User {
  final String id;
  final String email;
  final String? phone;
  final String status;
  final bool phoneVerified;
  final bool isOtpEnabled;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const User({
    required this.id,
    required this.email,
    this.phone,
    required this.status,
    required this.phoneVerified,
    required this.isOtpEnabled,
    this.createdAt,
    this.updatedAt,
  });

  factory User.fromJson(Map<String, dynamic> json) {
    return User(
      id: json['id'] as String,
      email: json['email'] as String,
      phone: json['phone'] as String?,
      status: json['status'] as String,
      phoneVerified: json['phone_verified'] as bool? ?? false,
      isOtpEnabled: json['is_otp_enabled'] as bool? ?? false,
      createdAt: json['created_at'] != null
          ? DateTime.tryParse(json['created_at'] as String)
          : null,
      updatedAt: json['updated_at'] != null
          ? DateTime.tryParse(json['updated_at'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'email': email,
      'phone': phone,
      'status': status,
      'phone_verified': phoneVerified,
      'is_otp_enabled': isOtpEnabled,
      'created_at': createdAt?.toIso8601String(),
      'updated_at': updatedAt?.toIso8601String(),
    };
  }

  User copyWith({
    String? id,
    String? email,
    String? phone,
    String? status,
    bool? phoneVerified,
    bool? isOtpEnabled,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return User(
      id: id ?? this.id,
      email: email ?? this.email,
      phone: phone ?? this.phone,
      status: status ?? this.status,
      phoneVerified: phoneVerified ?? this.phoneVerified,
      isOtpEnabled: isOtpEnabled ?? this.isOtpEnabled,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}
